website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,371 @@
+
+- **Unified Computation model** - an unified way to combine large batch style operations and frequent near real time streaming operations over a single unified dataset
+- **Self-Optimized Storage** - Automatically handle all the table storage maintenance such as compaction, clustering, vacuuming asynchronously and non-blocking to actual data changes
+- **Cloud Native Database** - abstracts Table/Schema from actual storage and ensures up-to-date metadata and indexes unlocking multi-fold read and write performance optimizations
+- **Engine neutrality** - designed to be neutral and not having a preferred computation engine. Apache Hudi will manage metadata, provide common abstractions and pluggable interfaces to most/all common computational engines.
+
+
+
+## Storage Format
+
+### Layout Hierarchy
+
+At a high level, Hudi organizes data into a high level directory structure under the base path (root directory for the Hudi table). The directory structure is based on coarse-grained partitioning values set for the dataset. Non-partitioned data sets store all the data files under the base path. Hudi storage format has a special reserved *.hoodie* directory under the base path that is used to store transaction logs and metadata.
+
+```
+/data/hudi_trips/ 					<== BASE PATH
+├── .hoodie/      					<== META BASE PATH
+│   └── metadata/ 
+├── americas/
+│   ├── brazil/
+│   │   └── sao_paulo/      <== PARTITIONED DIRECTORY 
+│   │       ├── <data_files>
+│   └── united_states/
+│       └── san_francisco/
+│           ├── <data_files>
+└── asia/
+    └── india/
+        └── chennai/
+            ├── <data_files>
+```
+
+Hudi storage format offers two table types offering different trade-offs between ingest and query performance and the data files are stored differently based on the chosen table type. 
+
+| Table Type    | Trade-off                                                    |
+| ------------- | ------------------------------------------------------------ |
+| Copy on Write | Optimized for read performance and ideal for slow changing datasets |
+| Merge-on-read | Optimized to balance the write and read performance and ideal for frequently changing datasets |
+
+
+
+### Data Model
+
+Within each partition, data is organized into key-value model. Every row is uniquely identified with a row key. To write a row into Hudi dataset, each row must specify the following user fields 
+
+| User fields                 | Description                                                  |
+| --------------------------- | ------------------------------------------------------------ |
+| Partitioning key [Optional] | Value of this field defines the directory hierarchy within the table base path. This essentially provides an hierarchy isolation for managing data and related metadata |
+| Row key(s)                     | Record keys uniquely identify a record/row within each partition if partitioning is enabled |
```

Comment:
better to call it "record key" everywhere?

Reply:
in case of global index or non-partition, then this is to identify a record across the whole dataset. just want to call out this case.